hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesState.java

```diff
@@ -396,16 +396,12 @@ private ServicesApp internalInit(
             app.initializationFlow().runWith(this, bootstrapProps);
             if (trigger == RESTART && isUpgrade) {
                 app.stakeStartupHelper().doUpgradeHousekeeping(networkCtx(), accounts(), stakingInfo());
-                // FIXME: uncomment after the merge of https://github.com/hashgraph/hedera-services/pull/5825
-                // (see https://github.com/hashgraph/hedera-services/issues/6037)
-                /*
-                if(getChild(StateChildIndices.ACCOUNTS) instanceof VirtualMap accounts) {
-                   accounts.fullRehash();
+                if (getChild(StateChildIndices.ACCOUNTS) instanceof VirtualMap<?, ?> accounts) {
+                    accounts.fullLeafRehash();
                 }
-                if(getChild(StateChildIndices.TOKEN_ASSOCIATIONS) instanceof VirtualMap tokenAssociations) {
-                    tokenAssociations.fullRehash();
+                if (getChild(StateChildIndices.TOKEN_ASSOCIATIONS) instanceof VirtualMap<?, ?> tokenAssociations) {
+                    tokenAssociations.fullLeafRehash();
```

Comment:
Why are only two virtual maps rehashed? There may be up to 5 of them: accounts, token rels, NFTs, file storage, and smart contracts.

Reply:
Good catch. I added rehashing for two other maps identified by `StateChildIndices.CONTRACT_STORAGE` and `StateChildIndices.STORAGE`

Reply:
And for `StateChildIndices.UNIQUE_TOKENS` too